server/middlewares/addResultsValidation.js

```diff
@@ -40,8 +41,11 @@ class addResultsValidation {
 
       if (Object.keys(fileDuplicates).length) return sendError(res, 422, ExcelValidators.fileDuplicateMessage(fileDuplicates));
 
+      // converts all data to lower case
+      results = Object.values(toLowerCase(results));
```

Comment:
I think this should come before `line 31` as we need to make the data lowercase before looking up `emailToUid` in the `Users` table.

We are storing emails in the table as lowercase but a user uploading may capitalize them.

Reply:
Good catch @jamsgra. Thanks man